### ISSUE TITLE
API count shows count of all aggregate tagged items and not count of tags

### ISSUE THREAD

**Issue by pm17788** [digitalocean/netbox#3239]
### Environment
* Python version: 3.5.2 
* NetBox version: 2.5.13

### Steps to Reproduce
1. Browse to: https://NetBoxURL/extras/tags, see count of tags (ex: 25)
2. Browse to: https://NetBoxURL/api/extras/tags, see count which is an aggregate of *all* tagged items, rather then the count of distinct tags in NetBox

### Expected Behavior
API count of tags should match the WebUI and the API

### Observed Behavior
API count of tags doesn't match the WebUI, or the behavior of `count` variable on other API endpoints.

**Comment by DanSheps** [comment]
Confirmed this is still present.

This looks to be to do with some performance optimizations made in the Pagination class.

**Comment by KhaledTo** [comment]
If my understanding is correct the count in ` /api/extras/tags` should show the number of distinct tags like in `/extras/tags`. 

I can take this, except if @pm17788 you want to work on this fix.

**Comment by pm17788** [comment]
@KhaledTo : Sadly, my Python-Fu is still weak. Still learning, so I'm not ready to go committing code yet.

**Comment by jeremystretch** [comment]
Looks like I fixed a slightly different problem under #3116.

FYI this has already been fixed under `develop-2.6`. The root issue stems from the use of `distinct=True` when annotating the TaggedItem count on the queryset. In v2.6, we've started using [subqueries](https://github.com/digitalocean/netbox/blob/develop-2.6/netbox/utilities/utils.py#L75) instead.

Not sure it makes sense to backport this utility function to the v2.5 train but we should be able to do a one-off implementation.
